### PR TITLE
Update build instructions

### PR DESCRIPTION
**Pull request by LWinterberg**

### CODE REVIEW

**Review by igorkorsukov — APPROVED** [review]

File: BUILDING.md [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
-* A CMake generator (eg Ninja)
-* A C++ compiler (MSVC on Windows)
+* A package manager (tested: Choco on Windows, homebrew on macOS)
+* A CMake generator (tested: Ninja)
+* A C++ compiler (tested: MSVC on Windows, g++ on Linux)
 * Qt 6.2.4, with modules:
-  * Qt Network Auth, 
-  * Qt 5 Compatibil
+  * Qt Network Authentification
+  * Qt 5 Compatibility
```
Comment (kryksyh) [inline BUILDING.md:14]:
```suggestion
  * Qt 5 Compatibility
  * Qt State Machines
```

Comment (kryksyh) [inline BUILDING.md:14]:
```suggestion
  * Qt Network Authorization
  * Qt State Machines
  * Qt 5 Compatibility Module
```

Comment (kryksyh) [reply BUILDING.md:14]: @LWinterberg can you also add update these lines?

Comment (kryksyh) [reply BUILDING.md:14]: @igorkorsukov sure, but it is required for Audacity as well
```
[cmake] CMake Error at _deps/muse_framework-src/buildscripts/cmake/FindQt6.cmake:57 (find_package):
[cmake]   By not providing "FindQt6StateMachine.cmake" in CMAKE_MODULE_PATH this
[cmake]   project has asked CMake to find a package configuration file provided by
[cmake]   "Qt6StateMachine", but CMake did not find one.
[cmake] 
[cmake]   Could not find a package configuration file provided by "Qt6StateMachine"
[cmake]   with any of the following names:
[cmake] 
[cmake]     Qt6StateMachineConfig.cmake
[cmake]     qt6statemachine-config.cmake
[cmake] 
[cmake]   Add the installation prefix of "Qt6StateMachine" to CMAKE_PREFIX_PATH or
[cmake]   set "Qt6StateMachine_DIR" to a directory containing one of the above files.
[cmake]   If "Qt6StateMachine" provides a separate development package or SDK, be
[cmake]   sure it has been installed.
[cmake] Call Stack (most recent call first):
[cmake]   CMakeLists.txt:158 (include)
```

Comment (igorkorsukov) [reply BUILDING.md:14]: oops, i'll fix it now

Comment (igorkorsukov) [reply BUILDING.md:14]: fixed

**Pull request merged**